### ISSUE TITLE
hv.Area not plotted if plotting limits smaller than value of Area

### ISSUE THREAD

**Issue by basnijholt** [ioam/holoviews#564]
```python
area = hv.Area((np.arange(10), np.ones(10)))
area
```
![download 6](https://cloud.githubusercontent.com/assets/6897215/13984445/92cafd08-f0f6-11e5-9be4-9bbcec2f9d18.png)

```python
area[:, 0:0.5]
```
![download 7](https://cloud.githubusercontent.com/assets/6897215/13984463/a009758a-f0f6-11e5-9e8f-61c4c430476b.png)

**Comment by philippjfr** [comment]
What were you expecting to happen? You sliced by the value dimension and all your values are one so you're left with an empty Area Element after slicing.

**Comment by basnijholt** [comment]
I would like to show a small tail, like in:
```
area = hv.Area((np.arange(10), 1-np.linspace(10)**2))
area[:, 0.5:1]
```

If you want I could post the actual image I'm trying to produce.

**Comment by philippjfr** [comment]
That would be helpful yes. The example you just posted does not work (the linspace signature is wrong). I'm also confused how you'd achieve a tail with an area that has a consistent height.

**Comment by basnijholt** [comment]
I think this illustrates it:
`gaps_plot[0:35, -0.05:0.2] + gaps_plot[0:35, -0.1:5]`
![download 2](https://cloud.githubusercontent.com/assets/6897215/13989842/67bb71bc-f112-11e5-9ac8-c11b4e0ef6f4.png)

**Comment by philippjfr** [comment]
It still looks like the expected behavior, by slicing the area you're dropping data points that fall outside this range, which is why the red ramp is dropped in the plot on the left. This goes back to the discussion in #181. The question is whether slicing should always slice the underlying data or whether it should just set the extents. Currently it does both where possible. So the correct solution in this case is to set only the extent not slice the underlying data.

If you think this behavior doesn't make sense you can reply to the issue above but this is not a bug.

**Comment by basnijholt** [comment]
If `extent` is the `.select` method, then this doesn't do what I want:
```
gaps_plot.select(E_k0=(-0.05, 0.2)))
```
<img width="338" alt="screen shot 2016-03-23 at 16 39 39" src="https://cloud.githubusercontent.com/assets/6897215/13990786/e96f0bda-f115-11e5-8591-9fd4a3b1f184.png">

**Comment by philippjfr** [comment]
That looks like the selection isn't being applied because ``E_k0`` doesn't match the actual dimension.

The ``extents`` parameter exists on every and allows setting the extents of the plot. When you slice using ``__getitem__`` or ``select`` (which are equivalent) two things happen, a) if possible it'll try to slice the underlying data and b) it sets the extents based on your current selection.

If you want to set just the extent you'll have to do so explicitly, e.g. by making a clone, ``gaps_plot.clone(extents=(None, -0.05, None, 0.2))``.

**Comment by basnijholt** [comment]
I think we can close this one as the issue is addressed in both #566 and #567

**Comment by philippjfr** [comment]
Good plan, closing.
